gem/lib/rbi-central/context.rb

```diff
@@ -0,0 +1,77 @@
+# typed: strict
+# frozen_string_literal: true
+
+module RBICentral
+  class Context
+    extend T::Sig
+    include CLI
+
+    sig { params(rbi_file: String, gem_name: String).void }
+    def initialize(rbi_file, gem_name)
+      @workdir = T.let(Dir.mktmpdir, String)
+      @rbi_file = rbi_file
+      @gem_name = gem_name
+      @gemfile = T.let(String.new, String)
+    end
+
+    sig { void }
+    def destroy!
+      FileUtils.rm_rf(@workdir)
+    end
+
+    sig { returns(T::Boolean) }
+    def run!
+      write_gemfile!
+
+      out, status = bundle_install!
+      unless status.success?
+        error("Can't install gem `#{@gem_name}` (#{out.gsub(/\n/, "")})")
+        return false
+      end
+
+      true
+    end
+
+    sig do
+      params(
+        name: String,
+        version: T.nilable(String),
+        github: T.nilable(String),
+        branch: T.nilable(String),
+        ref: T.nilable(String)
+      ).void
+    end
+    def add_gem_dependency(name, version: nil, github: nil, branch: nil, ref: nil)
+      @gemfile << "gem '#{name}'"
+      @gemfile << ", '#{version}'" if version
+      @gemfile << ", github: '#{github}'" if github
+      @gemfile << ", branch: '#{branch}'" if branch
+      @gemfile << ", ref: '#{ref}'" if ref
+      @gemfile << "\n"
+    end
+
+    private
+
+    sig { void }
+    def write_gemfile!
+      File.write("#{@workdir}/Gemfile", <<~GEMFILE)
+        source "https://rubygems.org"
+
+        #{@gemfile}
+      GEMFILE
+    end
+
+    sig { returns([String, Process::Status]) }
+    def bundle_install!
+      exec!("bundle config set --local path 'vendor/bundle'")
```

Comment:
Why install to `vendor/bundle`?

Reply:
To avoid depending on the user configuration. For example to avoid installing things in the system if the user is `root`.